src/main/java/net/javadiscord/javabot/listener/HugListener.java

```diff
@@ -24,6 +27,33 @@ public class HugListener extends ListenerAdapter {
 	private final AutoMod autoMod;
 	private final BotConfig botConfig;
 
+	private static final Pattern FUCKER = Pattern.compile("(fuck)(ing|er|k+)?", Pattern.CASE_INSENSITIVE);
+
+	private static String processFuck(String str) {
+		return FUCKER.matcher(str).replaceAll(matchResult -> {
+			String theFuck = matchResult.group(1);
+			String suffix = Objects.requireNonNullElse(matchResult.group(2), "");
+			String processedSuffix = switch(suffix.toLowerCase()) {
+				case "er", "ing" -> copyCase(suffix, 0, 'g') + suffix;
+				default -> suffix.toLowerCase().startsWith("k") ? copyCase(suffix, "g".repeat(suffix.length())) : "";
```

Comment:
I think it would make more sense to use use a distinct `case` for the empty `String`.
```suggestion
				case "" -> "";
				default -> copyCase(suffix, "g".repeat(suffix.length()));
```